src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -656,6 +676,49 @@ internal override bool IsCallerMemberName
             }
         }
 
+        internal override int CallerArgumentExpressionParameterIndex
+        {
+            get
+            {
+                const WellKnownAttributeFlags flag = WellKnownAttributeFlags.IsCallerArgumentExpression;
+
+                if (!_packedFlags.TryGetWellKnownAttribute(flag, out bool value))
+                {
+                    var discardedUseSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+                    bool isCallerArgumentExpression = !HasCallerLineNumberAttribute
+                        && !HasCallerFilePathAttribute
+                        && !HasCallerMemberNameAttribute
+                        && HasCallerArgumentExpressionAttribute
+                        && new TypeConversions(ContainingAssembly).HasCallerInfoStringConversion(this.Type, ref discardedUseSiteInfo);
+
+                    value = _packedFlags.SetWellKnownAttribute(flag, isCallerArgumentExpression);
+                }
+
+                if (value)
+                {
+                    foreach (var attribute in GetAttributes())
```

Comment:
>GetAttributes() [](start = 46, length = 15)

This call to `GetAttributes()` is expensive: it will cause to have to fully load attributes we wouldn't otherwise need to, which will cause memory pressure issues. Rather than loading all attributes, you're going to want to implement loading for just this attribute. Up to you whether you want to do it in this PR or just leave a prototype comment for now.

Here are some examples of how this works for `UnmanagedCallersOnly`:

[http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Metadata/PE/PEMethodSymbol.cs,1434](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Metadata/PE/PEMethodSymbol.cs,1434)
[http://sourceroslyn.io/#Microsoft.CodeAnalysis/MetadataReader/PEModule.cs,1140](http://sourceroslyn.io/#Microsoft.CodeAnalysis/MetadataReader/PEModule.cs,1140)

Reply:
@333fred Thanks. Will look into that. Meanwhile, does that mean there is a room for improvement for `NotNullIfParameterNotNull` as well?

https://github.com/dotnet/roslyn/blob/6da1274c9d24c2f90a48290394a951b23617f2a3/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs#L705-L724

Reply:
It does indeed.

Reply:
Opened #51995 for it, and updated my implementation here with c119c1b